tests/helpers/test_service.py

```diff
@@ -395,3 +394,36 @@ def test_async_get_all_descriptions(hass):
         mock_entities['light.kitchen'], mock_entities['light.living_room']]
     assert ('Not passing an entity ID to a service to target '
             'all entities is deprecated') in caplog.text
+
+
+async def test_require_admin_decorator(hass, hass_read_only_user,
+                                       hass_admin_user):
+    """Test the require_admin decorator."""
+    calls = []
+
+    @service.require_admin
+    async def mock_service(call):
+        calls.append(call)
+
+    hass.services.async_register('test', 'test', mock_service(hass))
```

Comment:
What will happen if developer didn't pass in hass? Do we clearly error message to remind developer the right syntax? 
```python
hass.services.async_register('test', 'test', mock_service)
```